Zend/zend_compile.c

```diff
@@ -903,6 +903,25 @@ uint32_t zend_add_class_modifier(uint32_t flags, uint32_t new_flag) /* {{{ */
 }
 /* }}} */
 
+uint32_t zend_add_anonymous_class_modifier(uint32_t flags, uint32_t new_flag)
+{
+	uint32_t new_flags = flags | new_flag;
+	if (new_flag & ZEND_ACC_EXPLICIT_ABSTRACT_CLASS) {
+		zend_throw_exception(zend_ce_compile_error,
+			"Cannot use the abstract modifier on an anonymous class", 0);
```

Comment:
It would be nice if the the new and the existing message were aligned to each other. For declared classes it says `Multiple abstract modifiers are not allowed`.

Reply:
Could you please clarify what you mean? :) I see two possible interpretations of what you said:
- Do you mean changing the message to "The Abstract modifier is not allowed on anonymous classes" ? (similarly for the final modifier)
- Or do you mean catching the case of using multiple abstract modifiers? Multiple abstract modifiers are not possible because the compilation will already throw an exception when a single abstract modifier is used.

I'm guessing it's the first one?

Reply:
@kocsismate Actually, the current error message is aligned. For example if you use "final + abstract" on a normal class you get "Cannot use the final modifier on an abstract class". So saying something like "Cannot use the abstract modifier on an anonymous class" does make sense imo.

Reply:
Oh yes, you are right!